### PR TITLE
Use Unforked React Native 0.60.6

### PR DESCRIPTION
**Pull request by NickGerleman**
Update package files + CI to have us always use stock React Native. We specify the exact version in package.json, since we could break if we accidentally rev our lockfile without realizing it.

CI here will not pass until #4015 is checked in and merged, but wanted to get this out early to get any feedback. I will be doing some manual testing of the integrated changes before merging this as well.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4017)

### CODE REVIEW

File: .ado/templates/build-rnw.yml [hunk 1]
```diff
@@ -49,8 +48,9 @@ steps:
       msbuildArguments:
         /p:PreferredToolArchitecture=${{parameters.preferredToolArchitecture}}
         /p:PlatformToolset=${{parameters.platformToolset}}
-        ${{parameters.msbuildArguments}}
         /p:BaseIntDir=$(BaseIntDir)
+        /p:PATCH_RN=${{parameters.applyRnPatches}}
```
Comment (NickGerleman) [inline .ado/templates/build-rnw.yml]: Want to rename `PATCH_RN` in a followup change to use normal conventions for properties. This was originally following the lead of OSS_RN but shouldn't be like this.

File: vnext/package.json [hunk 2]
```diff
@@ -52,13 +52,13 @@
...
     "just-scripts": "^0.24.2",
     "prettier": "1.17.0",
-    "react-native": "https://github.com/microsoft/react-native/archive/v0.60.0-microsoft.40.tar.gz",
+    "react-native": "0.60.6",
     "react": "16.8.6",
     "typescript": "3.5.3"
   },
   "peerDependencies": {
     "react": "16.8.6",
-    "react-native": "^0.60.0-0 || https://github.com/microsoft/react-native/archive/v0.60.0-microsoft.40.tar.gz"
```
Comment (kmelmon) [inline vnext/package.json]: >  "react-native": "^0.60.0-0 || https://github.com/microsoft/react-native/archive/v0.60.0-microsoft.40.tar.gz" [](start = 2, length = 110)

Why the additional -0 in ^0.60.0-0?

Comment (kmelmon) [inline vnext/package.json]: >https://github.com/microsoft/react-native/archive/v0.60.0-microsoft.40.tar.gz [](start = 34, length = 77)

Does Office still need this condition to run on the fork?

Comment (NickGerleman) [reply vnext/package.json]: This allows us to use arbitrary pre-release versions. E.g. The msft fork with vnext pre-release tag should work with this.

**Pull request merged**